HW_8/hw_8_2.js

```diff
@@ -0,0 +1,322 @@
+  let employeesNumber;
+  if (departmentOrEnterprise.hasOwnProperty('employees_count')) {
+    employeesNumber = departmentOrEnterprise.employees_count;
+  } else {
+    employeesNumber = getTotalEmployeesCount(departmentOrEnterprise);
+  }
+
+  if (/1$/.test(String(employeesNumber)) && employeesNumber !== 11) {
+    return departmentOrEnterprise.name !== "Администрация" ? `сотрудник` : `человек`;
+  } else if (/[2-4]$/.test(String(employeesNumber)) && !(/1[2-4]$/.test(String(employeesNumber)))) {
+    return departmentOrEnterprise.name !== "Администрация" ? `сотрудника` : `человека`;
+  } else {
+    return departmentOrEnterprise.name !== "Администрация" ? `сотрудников` : `человек`;
+  }
+}
+
+const getCompanyStructure = enterprisesArray => {
+  enterprisesArray.forEach((enterprise) => {
+    console.log(`${enterprise.name} (${getTotalEmployeesCount(enterprise)} ${getCorrectPeopleEnding(enterprise)})`);
+    for (const department of enterprise.departments) {
+      console.log(`- ${department.name} (${department.employees_count ? department.employees_count : `нет`} ${getCorrectPeopleEnding(department)})`);
+    }
+  });
+}
+  
+getCompanyStructure(enterprises);
+
+/*
+2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+Пример:
+getEnterpriseName(4) // Предприятие 1
+getEnterpriseName("Отдел маркетинга") // Предприятие 2
+*/
+console.log(`\nTask 2\n`);
+
+function getEnterpriseName (deptIdOrDeptName) {
+  const enterpriseName = searchEnterpriseName(deptIdOrDeptName);
+
+  const enterpriseNames = [];
+  for (const enterprise of enterpriseName) {
+    enterpriseNames.push(enterprise.name);
+  }
+  return enterpriseNames.join(` и `);
+}
+
+const searchEnterpriseName = deptIdOrDeptName => {
+  const foundArray = enterprises.filter(enterprise => enterprise.departments.find(department => department.name === deptIdOrDeptName || department.id === deptIdOrDeptName));
+  if (foundArray.length === 0) {
+    throw new Error (`No such enterprise with this search string '${deptIdOrDeptName}'.`);
+  }
+  return foundArray;
+}
+
+console.log(getEnterpriseName(4));
+console.log(getEnterpriseName("Отдел маркетинга"));
+
+/*
+3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+Пример:
+addEnterprise("Название нового предприятия")
+*/
+
+console.log(`\nTask 3\n`);
+
+function addEnterprise (name) {
+  enterprises.push({
+    id: getMaxId() + 1,
+    name,
+    departments: []
+  });
+  console.log(`Добавлено новое предприятие '${name}' с id ${getMaxId()}.`);
+}
+
+const getMaxId = () => {
+  let maxId = 0;
+  for (const enterprise of enterprises) {
+    if (maxId < enterprise.id) {
+      maxId = enterprise.id;
+    }
+    for (const department of enterprise.departments) {
+      if (maxId < department.id) {
+        maxId = department.id;
+      }
+    }
+  }
+  return maxId;
```

Comment:
Directly return the new ID so that you don't have to perform any further steps 
```suggestion
  return maxId + 1;
```